### PR TITLE
feat: Create shareable eslint configuration

### PR DESCRIPTION
**Pull request by florian-richter**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,8 +69,10 @@
     "compare-versions": "^3.6.0",
     "depcheck": "^1.0.0",
     "eslint": "^7.5.0",
+    "eslint-config-prettier": "^7.1.0",
```
Comment (FrankEssenberger) [inline package.json:72]: Question for the lerna pros. Should the new package not be included in the list of workspaces in the root package.json

**Review by FrankEssenberger — DISMISSED** [review]
Looks goods. Please consider my comment and also include a sentence in the release notes (CHANGELOG.md)

File: package.json (hunk 1 above)
Comment (marikaner) [reply package.json:72]: I think it should, so that we always use the local state

File: packages/eslint-config/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<a href="https://sap.com/s4sdk"><img src="https://help.sap.com/doc/2324e9c3b28748a4ae2ad08166d77675/1.0/en-US/logo-with-js.svg" alt="SAP Cloud SDK for JavaScript Logo" height="122.92" width="226.773"/></a>
+
+# @sap-cloud-sdk/eslint-config
+
+This package contains a shareable version of our `eslint` configuration.
```
Comment (marikaner) [inline packages/eslint-config/README.md:5]: Maybe you can give a bit more context - who is our. Imagine you would find this on npmjs.com.

**Review by marikaner — CHANGES_REQUESTED** [review]
I didn't check all rules, but it looks good to me ;) Please add one or two more sentences to the README. We will anyways have to improve and unify our readmes, but a bit more context could be helpful.

File: CHANGELOG.md [hunk 3]
```diff
@@ -95,9 +95,9 @@ Blog: TBD<br>
 ## Compatibility Notes
 
 - The methods `refreshTokenGrant`,`userTokenGrant` and `clientCredentialsGrant` of the `xsuaa-service` now accept a `string` or `XsuaaServiceCredentials` as first argument.
-In case it is a `string` it is treated as the URL to fetch the access token from.
-In case it is an object of type `XsuaaServiceCredentials`, the URL is built by appending `oauth/token` to the `.url` of the given object.
-In the past this suffix was also appended to inputs of type `string`.
+  In case it is a `string` it is treated as the URL to fetch the access token from.
```
[the remark this answers is not in the document]
Comment (florian-richter) [reply CHANGELOG.md:112]: Should I revert or keep it like this? I think the formatting done by prettier reads easier and should render identically.

[the remark this answers is not in the document]
Comment (marikaner) [reply CHANGELOG.md:112]: Aaaah ok, I saw it wrong. You didn't remove any leading -, right? If so no problemo.

**Review by marikaner — APPROVED** [review]
LGTM I would really like to try it before we release it for the first time...

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 
-[//]: # (Please don't delete the following comments and keep them in the beginning of this document. Also, keep the first line empty.)
+[//]: # "Please don't delete the following comments and keep them in the beginning of this document. Also, keep the first line empty."
```
Comment (marikaner) [inline CHANGELOG.md:2]: Why is this changed?

File: CHANGELOG.md [hunk 5]
```diff
@@ -18,7 +18,7 @@
 
 ## New Functionality
 
--
+- [eslint-config] Release a recommended, shareable `eslint` configuration based on the SAP Cloud SDKs internal configuration as `@sap-cloud-sdk/eslint-config`. Follow the [official guide](https://eslint.org/docs/developer-guide/shareable-configs#using-a-shareable-config) for information how to use the config.
```
Comment (marikaner) [inline CHANGELOG.md:21]:
```suggestion
- [eslint-config] Release a recommended, shareable `eslint` configuration based on the SAP Cloud SDK's internal configuration as `@sap-cloud-sdk/eslint-config`. Follow the [official guide](https://eslint.org/docs/developer-guide/shareable-configs#using-a-shareable-config) for information how to use the config.
```
```suggestion
- [eslint-config] Release a recommended, shareable `eslint` configuration based on the SAP Cloud SDKs internal configuration as `@sap-cloud-sdk/eslint-config`. Follow the [official guide](https://eslint.org/docs/developer-guide/shareable-configs#using-a-shareable-config) for information how to use the config.
```
